docs/production/modifying-zulip.md

```diff
@@ -0,0 +1,206 @@
+# Modifying or patching Zulip
+
+Zulip is 100% free and open source software, and you're welcome to
+modify it!  This page explains how to make and maintain modifications
+in a safe and convenient fashion.
+
+If you do modify Zulip and then report an issue you see in your
+modified version of Zulip, please be responsible about communicating
+that fact:
+
+* Ideally, you'd reproduce the issue in an unmodified version (e.g. on
+[chat.zulip.org](../contributing/chat-zulip-org.html) or
+[zulipchat.com](https://zulipchat.com)).
+* Where that is difficult or you think it's very unlikely your changes
+are related to the issue, just mention your changes in the issue report.
+
+## Making changes
+
+One way to modify Zulip is to just edit files under
+`/home/zulip/deployments/current` and then restart the server.  This
+can work OK for testing small changes to Python code or shell scripts.
+But we don't recommend this approach for maintaining changes because:
+
+* You cannot modify JavaScript, CSS, or other frontend files this way,
+  because they need to be compiled by Webpack, and we don't include
+  the uncompiled/source versions of these assets in our production
+  releases as they are very large and are not used at runtime.
```

Comment:
The uncompiled source actually is included in the source maps, although there isn’t a reasonable way to extract, edit, and recompile it. What’s not included, even if you had the patience to do that for some reason, is Webpack itself and its associated modules and configuration.